test/mpi/errors/spawn/connect_timeout.c

```diff
@@ -102,13 +102,21 @@ int test_mismatched_accept(MPI_Comm intra_comm, int gid)
 
         IF_VERBOSE(intra_rank == 0, ("client: connecting to <%s> with default timeout.\n", port));
         mpi_errno1 = MPI_Comm_connect(port, MPI_INFO_NULL, 0, intra_comm, &comm);
+        if (comm != MPI_COMM_NULL) {
+            IF_VERBOSE(intra_rank == 0, ("connection matched, freeing communicator.\n"));
+            MPI_Comm_free(&comm);
+        }
         if (mpi_errno1 != MPI_SUCCESS) {
```

Comment:
It looks like the code change does not match the commit message. Here you assume that as long as `comm` is changed, it will be a valid comm. Should we free `comm` only when `mpi_errno1` is `MPI_SUCCESS`?

Reply:
I updated the fix by setting `comm = MPI_COMM_NULL` before each call. We are not relying the failure behavior but relying on that a **successful** `MPI_Comm_{connect,accept}` will set `comm` to a *valid* handle.

Reply:
What if a failed `MPI_Comm_{connect,accept}` set `comm` to a garbage value?  I think the test itself should follow the standard, i.e., free comm only when the return errno is `MPI_SUCCESS`?

Reply:
> What if a failed `MPI_Comm_{connect,accept}` set `comm` to a garbage value? I think the test itself should follow the standard, i.e., free comm only when the return errno is `MPI_SUCCESS`?

I see. That makes sense. Will update

Reply:
Fixed.